src/samplers/pgibbs.jl

```diff
@@ -52,12 +52,12 @@ step(model::Function, spl::Sampler{PG}, vi::VarInfo, _::Bool) = step(model, spl,
 step(model::Function, spl::Sampler{PG}, vi::VarInfo) = begin
   particles = ParticleContainer{TraceR}(model)
 
-  vi.index = 0; vi.num_produce = 0;  # We need this line cause fork2 deepcopy `vi`.
+  vi.num_produce = 0;  # We need this line cause fork deepcopy `vi`.
```

Comment:
Could you expand on the comment "We need this line cause fork deepcopy `vi`."

Reply:
I'm not sure about that comment, but we need to make sure that `vi.num_produce = 0` before the `CSMC` sweep.